src/appshell/view/preferences/appearancepreferencesmodel.cpp

```diff
@@ -172,19 +235,17 @@ QString AppearancePreferencesModel::foregroundWallpaperPath() const
     return notationConfiguration()->foregroundWallpaperPath().toQString();
 }
 
-void AppearancePreferencesModel::setCurrentThemeIndex(int index)
+void AppearancePreferencesModel::setCurrentThemeCode(const QString& themeCode)
 {
-    ThemeList themes = allThemes();
-
-    if (index < 0 || index >= static_cast<int>(themes.size())) {
+    if (themeCode == currentThemeCode()) {
         return;
     }
 
-    if (index == currentThemeIndex()) {
-        return;
+    for (const ThemeInfo& theme : allThemes()) {
+        if (themeCode == QString::fromStdString(theme.codeKey)) {
+            uiConfiguration()->setCurrentTheme(theme.codeKey);
```

Comment:
It seems you missed break (or return) here

Reply:
Apologies, rookie mistake.
Fixed :)